File: fe/src/lib/types/daily-log.ts
export interface DailyLog {
    id: string;
    user_id: string;
    user_name: string;
    log_date: string;
    content: string;
    created_at: string;
    updated_at: string;
    links: DailyLogLink[];
}

export interface DailyLogLink {
    id: string;
    card_id: string;
    card_key: string;
    card_title: string;
}

export interface CreateDailyLogDto {
    log_date: string;
    content: string;
    card_ids?: string[];
}

export interface UpdateDailyLogDto {
    log_date?: string;
    content?: string;
    card_ids?: string[];
}

export interface ListDailyLogQuery {
    user_id?: string;
    start_date?: string;
    end_date?: string;
}

File: be/src/models/daily_log.rs
use chrono::{NaiveDate, NaiveDateTime};
use serde::{Deserialize, Serialize};
use sqlx::FromRow;
use uuid::Uuid;

#[derive(Debug, Serialize, Deserialize, FromRow, Clone)]
pub struct DailyLog {
    pub id: Uuid,
    pub user_id: Uuid,
    pub log_date: NaiveDate,
    pub content: String,
    pub created_at: NaiveDateTime,
    pub updated_at: NaiveDateTime,
}

#[derive(Debug, Serialize, Deserialize, FromRow, Clone)]
pub struct DailyLogWithUser {
    pub id: Uuid,
    pub user_id: Uuid,
    pub user_name: String,
    pub log_date: NaiveDate,
    pub content: String,
    pub created_at: NaiveDateTime,
    pub updated_at: NaiveDateTime,
}

#[derive(Debug, Serialize, Deserialize, FromRow, Clone)]
pub struct DailyLogLink {
    pub id: Uuid,
    pub daily_log_id: Uuid,
    pub card_id: Uuid,
    pub created_at: NaiveDateTime,
}
